src/viam/trajex/totg/trajectory.cpp

```diff
@@ -831,13 +877,21 @@ trajectory trajectory::create(class path p, options opt, integration_points poin
                     const auto [s_ddot_min, s_ddot_max] = compute_acceleration_bounds(
                         q_prime, q_double_prime, current_point.s_dot, traj.options_.max_acceleration, traj.options_.epsilon);
 
-                    if (s_ddot_min > s_ddot_max) [[unlikely]] {
+                    if (s_ddot_min - s_ddot_max > traj.options_.epsilon) [[unlikely]] {
                         throw std::runtime_error{"TOTG algorithm error: acceleration bounds are infeasible"};
                     }
 
+                    // Handle degenerate case where bounds are nearly equal (singularity/zero-acceleration point).
+                    // At such points, using the raw s_ddot_max might be slightly outside the feasible region
+                    // due to numerical errors.
+                    double s_ddot_to_use = s_ddot_max;
+                    if (s_ddot_max - s_ddot_min < traj.options_.epsilon) {
+                        s_ddot_to_use = std::min(s_ddot_min, s_ddot_max);
+                    }
```

Comment:
I suspect this correction could also be moved into `compute_acceleration_bounds`